### ISSUE TITLE
Supporting LayerNorm Quantization

### ISSUE THREAD

**Issue by ShinyCode** [google-ai-edge/ai-edge-torch#345]
### Description of the bug:

I am trying to run the ConvNeXt-tiny model on an edge device, which requires full int8 quantization. I found that the LayerNorm is not being quantized. Is there a way to force the quantization? I've even tried creating a custom annotator in `ai_edge_torch/quantize/pt2e_quantizer_utils.py` but despite annotating the node inputs/outputs fully, it remains unquantized in the final `.tflite.` Thanks!

### Actual vs expected behavior:

Right now, the LayerNorm is not quantized, which means that `Dequantize` and `Quantize` nodes are still present in the graph. The expected behavior is that no such nodes are present, allowing the computation to run fully in int8.

### Any other information you'd like to share?

These are the package versions I am using:
```
ai-edge-litert-nightly==1.0.1.dev20241029                                           
ai-edge-model-explorer==0.1.14
ai-edge-model-explorer-adapter==0.1.5
ai-edge-quantizer-nightly==0.0.1.dev20241030                                       
ai-edge-torch-nightly==0.3.0.dev20241029
```
Here is a simple model that replicates this behavior:

```python
class Model(nn.Module):
    def __init__(self):
        super().__init__()
        self.proj1 = nn.Linear(128, 64)
        self.norm = nn.LayerNorm(64)
        self.proj2 = nn.Linear(64, 32)

    def forward(self, x):
        x = self.proj1(x)
        x = self.norm(x)
        x = self.proj2(x)
        return x
```

The resulting graph is shown here, which is suboptimal since it has Dequantize and Quantize, which means that data would be shuffled to-and-from the edge device.

![image](https://github.com/user-attachments/assets/41fa43a5-a4e4-4a36-82dd-01c12a70e557)

I've attached the snippet of code that reproduces this:
```python
# Instantiate the dummy model
model = Model()
model.eval()

# Create sample inputs
sample_inputs = (torch.randn(32, 128),)
model(sample_inputs[0])

# Create quantizer
qconfig = pt2e_quantizer.get_symmetric_quantization_config(
    is_per_channel=False,
    is_qat=False,
    is_dynamic=False,
)
quantizer: pt2e_quantizer.PT2EQuantizer
quantizer = pt2e_quantizer.PT2EQuantizer().set_global(
    qconfig,
)
pt2e_torch_model = torch._export.capture_pre_autograd_graph(model, sample_inputs)
pt2e_torch_model = quantize_pt2e.prepare_pt2e(pt2e_torch_model, quantizer)

# Run an input to trace the graph
pt2e_torch_model(sample_inputs[0])

# Convert and export model
pt2e_torch_model = quantize_pt2e.convert_pt2e(pt2e_torch_model, fold_quantize=False)
pt2e_drq_model = ai_edge_torch.convert(
    pt2e_torch_model,
    sample_inputs,
    quant_config=quant_config.QuantConfig(pt2e_quantizer=quantizer),
)
```

**Comment by ShinyCode** [comment]
Just adding on, this is the code I've been trying to write... It looks like all the Nodes are being annotated, but I'm still ending up with the `Dequantize` and `Quantize` nodes in the final `.tflite`.
```python
@register_annotator("layer_norm")
def _annotate_layer_norm(
    gm: torch.fx.GraphModule,
    quantization_config: Optional[QuantizationConfig],
    filter_fn: Optional[Callable[[Node], bool]] = None,
) -> Optional[List[List[Node]]]:
  annotated_partitions = []
  for n in gm.graph.nodes:
    if n.op != "call_function" or n.target not in [
        torch.ops.aten.layer_norm.default,
    ]:
      continue
    norm_node = n

    input_qspec_map = {}
    input_act = norm_node.args[0]
    assert isinstance(input_act, Node)
    input_qspec_map[input_act] = get_input_act_qspec(quantization_config)

    weight1 = norm_node.args[2]
    assert isinstance(weight1, Node)
    input_qspec_map[weight1] = get_weight_qspec(quantization_config)

    weight2 = norm_node.args[3]
    assert isinstance(weight2, Node)
    input_qspec_map[weight2] = get_weight_qspec(quantization_config)

    # adding weight node to the partition as well
    partition = [norm_node, norm_node.args[2], norm_node.args[3]]

    if _is_annotated(partition):
      continue

    if filter_fn and any(not filter_fn(n) for n in partition):
      continue

    norm_node.meta["quantization_annotation"] = QuantizationAnnotation(
        input_qspec_map=input_qspec_map,
        output_qspec=get_output_act_qspec(quantization_config),
        _annotated=True,
    )
    _mark_nodes_as_annotated(partition)
    annotated_partitions.append(partition)
  return annotated_partitions
  ```

**Comment by pkgoogle** [comment]
Hi @ShinyCode, I was able to make it work with tfl quantization: https://ai.google.dev/edge/litert/models/post_training_quantization#integer_only

my code:
```py
import torch
from torch import nn
import tensorflow as tf
import ai_edge_torch
import numpy as np


class Model(nn.Module):
    def __init__(self):
        super().__init__()
        self.proj1 = nn.Linear(128, 64)
        self.norm = nn.LayerNorm(64)
        self.proj2 = nn.Linear(64, 32)

    def forward(self, x):
        x = self.proj1(x)
        x = self.norm(x)
        x = self.proj2(x)
        return x


model = Model()
model.eval()

sample_inputs = (torch.randn(32, 128),)

def representative_dataset():
    for _ in range(100):
      data = np.random.rand(32, 128)
      yield [data.astype(np.float32)]
 

# Pass TfLite Converter quantization flags to _ai_edge_converter_flags parameter.
tfl_converter_flags = {
    'optimizations': [tf.lite.Optimize.DEFAULT],
    'representative_dataset': representative_dataset,
    'target_spec.supported_ops': [tf.lite.OpsSet.TFLITE_BUILTINS_INT8],
    'inference_input_type': tf.int8,
    'inference_output_type': tf.int8,
}

tfl_fullint_model = ai_edge_torch.convert(
    model, sample_inputs, _ai_edge_converter_flags=tfl_converter_flags
)

tfl_fullint_model.export('layernorm.tflite')
```

Can you try this out and let me know if it works?

**Comment by ShinyCode** [comment]
Hi @pkgoogle thanks for your help! It worked on our side.

The only further change we needed (just posting this here in case it benefits others) is since we needed per-tensor quantization, we added the `_experimental_disable_per_channel` flag.

```python
tfl_converter_flags = {
    "optimizations": [tf.lite.Optimize.DEFAULT],
    "representative_dataset": partial(
        representative_dataset, dataset_path, dataset
    ),
    "target_spec.supported_ops": [tf.lite.OpsSet.TFLITE_BUILTINS_INT8],
    "inference_input_type": tf.int8,
    "inference_output_type": tf.int8,
    "_experimental_disable_per_channel": True  # Add this flag
}
```
